packages/ui-tooltip/src/Tooltip/props.ts

```diff
@@ -31,127 +31,177 @@ import { element } from '@instructure/ui-prop-types'
 import type {
   PropValidators,
   AsElementType,
-  TooltipTheme
+  TooltipTheme,
+  OtherHTMLAttributes
 } from '@instructure/shared-types'
 import type {
   PlacementPropValues,
   PositionConstraint,
   PositionMountNode
 } from '@instructure/ui-position'
 import type { WithStyleProps, ComponentStyle } from '@instructure/emotion'
+import type { PopoverOwnProps } from '@instructure/ui-popover'
 
 type TooltipRenderChildrenArgs = {
   focused: boolean
-  getTriggerProps: <P extends Record<string, any>>(
-    props: P
-  ) => { 'aria-describedby': string } & P
+  getTriggerProps: <TriggerProps extends Record<string, any>>(
+    props: TriggerProps
+  ) => { 'aria-describedby': string } & TriggerProps
 }
 
 type TooltipRenderChildren = (
   args: TooltipRenderChildrenArgs
 ) => React.ReactNode
 
 type TooltipOwnProps = {
-  renderTip: React.ReactNode | ((...args: any[]) => any)
-  children: React.ReactNode | TooltipRenderChildren
-  isShowingContent?: boolean
-  defaultIsShowingContent?: boolean
-  as?: AsElementType
-  on?: ('click' | 'hover' | 'focus') | ('click' | 'hover' | 'focus')[]
-  color?: 'primary' | 'primary-inverse'
-  placement?: PlacementPropValues
-  mountNode?: PositionMountNode
-  constrain?: PositionConstraint
-  positionTarget?: PositionMountNode
-  offsetX?: string | number
-  offsetY?: string | number
-  onShowContent?: (...args: any[]) => any
-  onHideContent?: (...args: any[]) => any
-}
-
-type PropKeys = keyof TooltipOwnProps
-
-type AllowedPropKeys = Readonly<Array<PropKeys>>
-
-// For now it doesn't need the OtherHTMLAttributes, because the extra props
-// get passed to Popover and it doesn't handle them
-type TooltipProps = TooltipOwnProps & WithStyleProps<TooltipTheme, TooltipStyle>
-
-type TooltipStyle = ComponentStyle<'tooltip'>
-
-const propTypes: PropValidators<PropKeys> = {
   /**
-   * @param {Object} renderProps
-   * @param {Boolean} renderProps.focused - Is the Tooltip trigger focused?
-   * @param {Function} renderProps.getTriggerProps - Props to be spread onto the trigger element
+   * A ReactNode or a function that returns a ReactNode with the following params:
+   *
+   * @param {Boolean} focused - Is the Tooltip trigger focused?
+   * @param {Function} getTriggerProps - Props to be spread onto the trigger element
    */
-  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
+  children: React.ReactNode | TooltipRenderChildren
+
```

Comment:
Not something that is concerning this PR, but JSDoc comments for TS types (PropTypes too for the matter) does not render nicely in our `Properties` table:
<img width="1011" alt="Screenshot 2022-01-03 at 10 07 02" src="https://user-images.githubusercontent.com/20640237/147914466-b95916a8-632a-41bc-8938-3abb4787494c.png">

Maybe it would be a good idea to investigate this later?

Reply:
Yeah, well, this seems to have been like this forever, it is the same in V7 :/ Also, I think that once we have better control over our TS types in the docs, and can resolve stuff like "TooltipRenderChildren" syntax, then the JSDocs won't be needed at all.